latitude_inventory.py

```diff
@@ -174,21 +187,59 @@ def get_servers(self) -> list[Server]:
             all_servers += servers
             page += 1
         return all_servers
+    
+    def get_tags(self) -> list[str]:
+        latitude_api_token = self.get_option("latitude_api_token")
+        params = {
+            "sort": "id",
+        }
 
-    def add_sever(self, server: Server) -> None:
+        url = "https://api.latitude.sh/tags"
+        headers = {
+            "accept": "application/json",
+            "Authorization": latitude_api_token,
+        }
+        response = requests.get(url, headers=headers, params=params, timeout=10)
+        response.raise_for_status()
+        tags = response.json().get("data", [])
+        all_tag_names = [t["attributes"]["name"] for t in tags]
+        return all_tag_names
+        
+
+    def add_sever(self, inventory: InventoryData, server: Server) -> None:
         server_attributes = server["attributes"]
         hostname = server_attributes["hostname"]
         # server_type on latitude, see https://docs.latitude.sh/reference/get-plans. Examples: "c2-small-x86", "c3-medium-arm"
         server_type = server_attributes["plan"]["slug"]
-        group = self.get_hosts_group(hostname)
+        tags = [t["name"] for t in server_attributes["tags"]]
+        role_tags = [t for t in tags if t.startswith("role_")]
+        if role_tags:
+            groups = [t[len("role_"):] for t in role_tags]
+        else:   
+            groups = [self.get_hosts_group(hostname)]
+
+        include_tags = self.get_option("include_tags")
+        exclude_tags = self.get_option("exclude_tags")
+
+        if include_tags is not None:
+            if not any(tag in include_tags for tag in tags):
+                return
+
+        if exclude_tags is not None:
+            if any(tag in exclude_tags for tag in tags):
+                return
 
-        self.inventory.add_host(hostname, group=group)
+        for group in groups:
+            if group:
+                self.inventory.add_host(hostname, group=group)
+            else:
+                self.inventory.add_host(hostname)
 
         host_vars = {}
         host_vars["public_ip_address"] = server_attributes["primary_ipv4"]
         host_vars["server_name"] = hostname
         host_vars["server_type"] = server_type
-        host_vars["group"] = group
+        host_vars["tags"] = [t["name"] for t in server_attributes["tags"]]
```

Comment:
```suggestion
        host_vars["tags"] = tags
```
you already have tags from line 214